plugins/modules/vultr_block_storage.py

```diff
@@ -216,28 +242,140 @@ def absent_block_storage_volume(self):
                 )
         return volume
 
+    def detached_block_storage_volume(self):
+        volume = self.present_block_storage_volume()
+        if volume.get('attached_to_SUBID') is None:
+            return volume
+
+        self.result['changed'] = True
+
+        if not self.module.check_mode:
+            data = {
+                'SUBID': volume['SUBID'],
+                'live': self.module.params.get('live_attachment')
```

Comment:
as this should be a string of `yes` or `no` (thanks vultr...) and `live_attachment` is bool, you can use the helper method https://github.com/ngine-io/ansible-collection-vultr/blob/master/plugins/module_utils/vultr.py#L125

```
'live': self.get_yes_or_no('live_attachment')
```

Reply:
fixed